packages/dev-frontend/src/components/Trove/Adjusting.tsx

```diff
@@ -48,18 +48,75 @@ const feeFrom = (original: Trove, edited: Trove, borrowingRate: Decimal): Decima
   }
 };
 
+const applyUnsavedCollateralChanges = (unsavedChanges: Difference, trove: Trove) => {
+  if (unsavedChanges.absoluteValue) {
+    return unsavedChanges.positive
+      ? trove.collateral.add(unsavedChanges.absoluteValue)
+      : unsavedChanges.negative
+      ? trove.collateral.sub(unsavedChanges.absoluteValue)
```

Comment:
The unsaved change can be negative with greater absolute value than the Trove's new collateral. I guess the best we can do is clamp the edited collateral to 0.